fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/SplitAggregateRule.java

```diff
@@ -118,8 +116,10 @@ private boolean mustGenerateMultiStageAggregate(OptExpression input, List<CallOp
 
     // Note: This method logic must consistent with CostEstimator::needGenerateOneStageAggNode
     private boolean needGenerateMultiStageAggregate(OptExpression input, List<CallOperator> distinctAggCallOperator) {
-        // 1. check if can generate multi stage aggregate.
-        if (!canGenerateMultiStageAggregate(input)) {
+        // 1. Must do one stage aggregate If the child contains limit,
+        //    the aggregation must be a single node to ensure correctness.
+        //    eg. select count(*) from (select * table limit 2) t
+        if (input.inputAt(0).getOp().hasLimit()) {
```

Comment:
is it possible to have child without limit but grandchild with limit?

Reply:
The rule is for  agg->leaf pattern. I think we only consider child is OK.